tests/test_class_attributes.rs

```diff
@@ -235,6 +235,26 @@ fn test_renaming_all_struct_fields() {
     });
 }
 
+#[pyclass(get_all, set_all, new = "from_fields")]
+struct AutoNewCls {
+    a: i32,
+    b: String,
+    c: Option<f64>,
+}
```

Comment:
It might be nice to add a UI test of what happens if a field it not convertible from Python, e.g.

```rust
#[pyclass(new = "from_fields")]
struct NonPythonField {
    field: Box<dyn std::error::Error>
}
```

Reply:
added in https://github.com/PyO3/pyo3/pull/5421/commits/c2dfd2e3fe893e7611217d531f0558b24384e37c

the error is a little bit big but the top line says what it actually is